providers/tests/system/amazon/aws/example_sagemaker_unified_studio.py

```diff
@@ -0,0 +1,149 @@
+"""
+
+pytestmark = pytest.mark.skipif(
+    not AIRFLOW_V_2_10_PLUS, reason="Test requires Airflow 2.10+"
+)
+
+DAG_ID = "example_sagemaker_unified_studio"
+
+# Externally fetched variables:
+DOMAIN_ID_KEY = "DOMAIN_ID"
+PROJECT_ID_KEY = "PROJECT_ID"
+ENVIRONMENT_ID_KEY = "ENVIRONMENT_ID"
+S3_PATH_KEY = "S3_PATH"
+
+sys_test_context_task = (
+    SystemTestContextBuilder()
+    .add_variable(DOMAIN_ID_KEY)
+    .add_variable(PROJECT_ID_KEY)
+    .add_variable(ENVIRONMENT_ID_KEY)
+    .add_variable(S3_PATH_KEY)
+    .build()
+)
+
+
+@task
+def emulate_mwaa_environment(
+    domain_id: str, project_id: str, environment_id: str, s3_path: str
+):
+    """
+    Sets several environment variables in the container to emulate an MWAA environment provisioned
+    within SageMaker Unified Studio.
+    """
+    AIRFLOW_PREFIX = "AIRFLOW__WORKFLOWS__"
+    os.environ[f"{AIRFLOW_PREFIX}DATAZONE_DOMAIN_ID"] = domain_id
+    os.environ[f"{AIRFLOW_PREFIX}DATAZONE_PROJECT_ID"] = project_id
+    os.environ[f"{AIRFLOW_PREFIX}DATAZONE_ENVIRONMENT_ID"] = environment_id
+    os.environ[f"{AIRFLOW_PREFIX}DATAZONE_SCOPE_NAME"] = "dev"
+    os.environ[f"{AIRFLOW_PREFIX}DATAZONE_STAGE"] = "prod"
+    os.environ[f"{AIRFLOW_PREFIX}DATAZONE_ENDPOINT"] = (
+        "https://datazone.us-east-1.api.aws"
+    )
+    os.environ[f"{AIRFLOW_PREFIX}PROJECT_S3_PATH"] = s3_path
+    os.environ[f"{AIRFLOW_PREFIX}DATAZONE_DOMAIN_REGION"] = "us-east-1"
+
+
+with DAG(
+    DAG_ID,
+    schedule="@once",
+    start_date=datetime(2021, 1, 1),
+    tags=["example"],
+    catchup=False,
+) as dag:
+    test_context = sys_test_context_task()
+
+    test_env_id = test_context[ENV_ID_KEY]
+    domain_id = test_context[DOMAIN_ID_KEY]
+    project_id = test_context[PROJECT_ID_KEY]
+    environment_id = test_context[ENVIRONMENT_ID_KEY]
+    s3_path = test_context[S3_PATH_KEY]
+
+    setup_mock_mwaa_environment = emulate_mwaa_environment(
+        domain_id,
+        project_id,
+        environment_id,
+        s3_path,
+    )
+
+    # [START howto_operator_sagemaker_unified_studio_notebook]
+    notebook_path = "test_notebook.ipynb"  # This should be the path to your .ipynb, .sqlnb, or .vetl file in your project.
+
+    run_notebook = SageMakerNotebookOperator(
+        task_id="initial",
+        input_config={"input_path": notebook_path, "input_params": {}},
+        output_config={"output_formats": ["NOTEBOOK"]},  # optional
+        compute={
+            "InstanceType": "ml.m5.large",
+            "VolumeSizeInGB": 30,
+        },  # optional
+        termination_condition={"MaxRuntimeInSeconds": 600},  # optional
+        tags={},  # optional
+        wait_for_completion=True,  # optional
+        poll_interval=5,  # optional
+        deferrable=False,  # optional
+    )
+    # [END howto_operator_sagemaker_unified_studio_notebook]
+
+    chain(
+        # TEST SETUP
+        test_context,
+        setup_mock_mwaa_environment,
+        # TEST BODY
+        run_notebook,
+    )
```

Comment:
No teardown needed? What if things break, get stuck, timeout? Any way to manually stop a noteboook execution?

Reply:
Since we're just testing the execution of the training job, we don't have any resources we'll need to teardown. The training job has a MaxRuntimeSeconds so it will auto-time-out if things hang. We also don't want to delete the training job during teardown in case we need it to investigate a test failure.